Tests/DependencyInjection/Compiler/Builder/DefinitionBuilderTest.php

```diff
@@ -61,6 +62,14 @@ public function testGetHandlerDefinition()
         $this->assertInstanceOf(Expression::class, $customMiddlewareExpression);
         $this->assertSame('service("test_middleware").attach()', $customMiddlewareExpression->__toString());
 
+        $customMiddlewareCall = array_shift($methodCalls);
```

Comment:
I know it was done for all previous middleware tests as well, but I think it's worth to get rid of the `array_shift` hell.

I though about sth like:
```php
// define expected middleware calls
$expectedMiddlewareCalls = [
    'service("test_middleware").attach()' => 'push',
    'service("test_middleware").attach2()' => 'push',
    'service("guzzle_bundle.middleware.log.clientName").log()' => 'push',
    'service("guzzle_bundle.middleware.event_dispatch.clientName").dispatch()' => 'unshift',
];

$doneMiddlewareCalls = [];
foreach ($methodCalls as $methodCall) {
    /** @var Expression[] $customMiddlewareExpressions */
    $customMiddlewareExpressions = $methodCall[1];
    $customMiddlewareExpression = $customMiddlewareExpressions[0];
    $this->assertInstanceOf(Expression::class, $customMiddlewareExpression);

    $doneMiddlewareCall = $customMiddlewareExpression->__toString();

    $this->assertSame($expectedMiddlewareCalls[$doneMiddlewareCall], $methodCall[0]);

    $doneMiddlewareCalls[] = $doneMiddlewareCall;
}

$this->assertCount(0, array_diff(array_keys($expectedMiddlewareCalls), $doneMiddlewareCalls));
```

What do you think @wizbit ?
It would makes it easy to add more system defined `Middleware`s later on and the code is more readable since it's shorter.